docs/openapi_spec.json

```diff
@@ -0,0 +1,212 @@
+        "responses": {
+          "200": {
+            "description": "An array of tuples with an identifier for the student and the name of the team that they were allocated to",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                    "items": {
+                      "oneOf": [{
+                      "$ref": "#/components/schemas/StudentInTeam"
+                    }]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "Student": {
+        "type": "object",
+        "properties": {
+          "firstName": {
+            "type": "string"
+          },
+          "lastName": {
+            "type": "string"
+          },
+          "email": {
+            "type": "string"
+          },
+          "tumId": {
+            "type": "string"
+          },
+          "gender": {
+            "$ref": "#/components/schemas/Gender"
+          },
+          "nationality": {
+            "type": "string"
+          },
+          "major": {
+            "type": "string"
+          },
+          "semester": {
+            "type": "number"
+          },
+
+          "germanLanguageLevel": {
+            "type": "string"
+          },
+          "englishLanguageLevel": {
+            "type": "string"
+          },
+
+          "iosDev": {
+            "type": "string"
+          },
+          "appStoreLink": {
+            "type": "string"
+          },
+          "iOSDevExplained": {
+            "type": "string"
+          },
+
+          "introAssessment": {
+            "type": "string"
+          },
+          "introAssessmentTutor": {
+            "type": "string"
+          },
+          "devices": {
+            "type": "array",
+            "items": {
+              "oneOf": [ {
+                "$ref": "#/components/schemas/Device"
+              }]
+            }
+          },
+          "skills": {
+            "type": "array",
+            "items": {
+              "oneOf": [ {
+                "$ref": "#/components/schemas/Skill"
+              }]
+            }
+          },
+          "otherSkills": {
+            "type": "string"
+          },
+          "teamPriorities": {
+            "type": "array",
+            "items": {
+              "oneOf": [{
+                "$ref": "#/components/schemas/Team"
+              }]
+            }
+          },
+
+          "studentComments": {
+            "type": "string"
+          },
+          "supervisorRating": {
+            "$ref": "#/components/schemas/SkillLevel"
+          },
+
+          "tutorComments": {
+            "type": "string"
+          }
+        }
+      },
+      "Team": {
+        "type": "string"
+      },
+      "Device": {
+        "type": "string",
+        "enum": ["Iphone", "Ipad", "Mac", "Watch", "IphoneAR", "IpadAR"]
+      },
+      "Skill": {
```

Comment:
I like this suggested refactoring. Just to clarify, the model previously contained an interest level and "justification",  for a skill, I am simplifying this now so that a skill just has a self reported skill level (in addition to a title and description like you listed), nothing else. The interest level by TEASE was used to be able to highlight students that had a particular interest in a certain skill, but in my opinion experience/expertise is a lot more important.

Reply:
Now thinking about it again, should skills really only have a description field where they can explain what the skill is? Or should we leave a second field where they can justify or explain their self assessment for skill level?